modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -584,6 +584,19 @@ Mat::Mat(const std::initializer_list<_Tp> list)
         return;
     Mat((int)list.size(), 1, traits::Type<_Tp>::value, (uchar*)list.begin()).copyTo(*this);
 }
+
+template<typename _Tp, typename> inline
+Mat::Mat(const std::initializer_list<int> sizes, const std::initializer_list<_Tp> list)
+    : Mat()
+{
+    size_t size_total = 1;
+    int *sz = (int*)sizes.begin();
+    for(int i = 0; i < sizes.size(); i++)
+        size_total *= sz[i];
+    CV_Assert(list.size() != 0 || size_total != list.size());
```

Comment:
This condition is always true for any on-empty list.
Also "size_total != list.size()" is wrong initially.

Reply:
I made a mistake. 
What I want to do was not to generate an assertion if the condition of "size_total == list.size()" is satisfied. I fix this code soon.